### DISCUSSION TITLE
Any way to reflect on the collections?

### DISCUSSION THREAD

**Discussion by square-foot** [pocketbase/pocketbase#3319] [Q&A]
Hi
I am new to Pocketbase. Can someone help me with a function that would reflect on the structure of any collection and explain each field in there; including the field options that were set; and in case of "select" type; what were the string options available for that field; and in case of "relation" type, which collection was being referred? etc.

I am making a small attempt to make a code generator that would help write a rough outline of the code needed for a CRUD type SaaS ... hence ideally this kind of reflection capabilities would help

Thanks

**Comment by ganigeorgiev** [comment]
I'm not entirely sure that I understand what you are looking for, but you can check:
- (Go) https://pocketbase.io/docs/go-collections/
- (JS `pb_hooks`) https://pocketbase.io/docs/js-collections/
- (web APIs) https://pocketbase.io/docs/api-collections/

**Comment by square-foot** [comment]
Thank you very much. 

I got the answer here: https://pocketbase.io/docs/js-collections/ 

I have an hazy idea of using the excellent admin interface that the user can use to create the collections, and then with one special "route" it can export all the collection schema into a special format that my code generator can then work on to automatically write out some of the code at least.  So the Goja JS that I write for the route would need to reflect on what the collections  were defined, etc. (that is what I meant by "reflection" ... not sure if that's the right term) and I got what I was looking for. Thank you very much. Grateful!

**Reply by ganigeorgiev** [reply]
_Note that if you just need access to the collection structure you may not need a custom route._
_The existing [`GET /api/collections/:collectionIdOrName`](https://pocketbase.io/docs/api-collections/#view-collection) already returns a structured JSON with all fields of the collection._

**Reply by square-foot** [reply]
Thanks. That is useful to know!
